shell.nix

```diff
@@ -1,20 +1,29 @@
-# Entering the shell for `proto3-suite` requires rebuilding `proto3-suite`
-# when basically _any_ file changes, which makes development painful.
+# Entering the shell for `proto3-suite` requires rebuilding `proto3-suite-boot`
+# non-incrementally when basically _any_ file changes, which makes development
+# painful.
 #
 # If you don't need to run the test suite, you can forgo test dependencies
-# and skip `compile-proto-file` codegen by running:
+# and skip automatic `compile-proto-file` codegen by running:
 #
 #   $ nix-shell --arg fast true
 #
 { fast ? false
 , compiler ? "ghc884"
 , enableDhall ? false
 , enableSwagger ? true
-}@args:
+}:
 
 with (import ./default.nix { inherit compiler enableDhall enableSwagger; });
 
 if fast then
-  proto3-suite-boot.env
+  builtins.trace
+    ''
+    Using `fast` shell... here are some gotchas to be aware of:
+      - You must run codegen for test protos manually using the
+        `compile-test-protos` shell script (in the shell's `PATH`)
+      - You can _build_ the test suite, but you can't _run_ it (without it failing
+        due to missing dependencies)
+    ''
```

Comment:
I admit I'm a bit confused about this bit.

You have two bullet points, the first saying that you *must* manually run a command for the test suite to work, then the second says that the test suite *doesn't* work.

Could possibly use a rewording?

Reply:
Yeah you're right, that isn't clear. You need to run that script to create Haskell modules required for building the test suite. But that's as far as you can go; you can't actually run it without it failing.

Reply:
I guess I'm not entirely sure why you'd want to build the test suite if you're not going to run it.

Reply:
Being able to build the test suite isn't a huge benefit if you can't run it, but I figured it was still an improvement.

Reply:
Reworded things, hopefully it's clearer now. Ultimately I want to get rid of the bootstrapping process, and then the "fast" shell won't be necessary. But for now, I'm providing a "fast" shell and making the caveats clear.